sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -348,28 +348,41 @@ case class InsertIntoTable(
     partition: Map[String, Option[String]],
     child: LogicalPlan,
     overwrite: Boolean,
-    ifNotExists: Boolean)
+    ifNotExists: Boolean,
+    options: Map[String, String])
```

Comment:
Right now, yes. But I changed from a boolean to a map because there are other options that will be added, like `filesPerPartition`. Changing the case class requires code changes in several places just to add `_` to a match expression or pass the argument on. Using a map means we can extend InsertIntoTable without updating the signature all the time. Patches should be smaller and have fewer conflicts.